rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java

```diff
@@ -615,15 +615,7 @@ private <T> List<JRightPadded<T>> visitRightPadded(ASTNode[] nodes, @Nullable St
         }
 
         private Expression insideParentheses(ASTNode node, Function<Space, Expression> parenthesizedTree) {
-            Integer insideParenthesesLevel;
-            Object rawIpl = node.getNodeMetaData("_INSIDE_PARENTHESES_LEVEL");
-            if(rawIpl instanceof AtomicInteger) {
-                // On Java 11 and newer _INSIDE_PARENTHESES_LEVEL is an AtomicInteger
-                insideParenthesesLevel = ((AtomicInteger) rawIpl).get();
-            } else {
-                // On Java 8 _INSIDE_PARENTHESES_LEVEL is a regular Integer
-                insideParenthesesLevel = (Integer) rawIpl;
-            }
+            Integer insideParenthesesLevel = getInsideParenthesesLevel(node);
```

Comment:
This is where we still see a difference between Groovy 2.5 and 3+;
- on 3+ the ConstantExpression has a metaDataMap containing `_INSIDE_PARENTHESES_LEVEL: 1`
- on 2.5 the ConstantExpression lacks a metaDataMap

We then either enter (for 3+) the below `if`, or skip it for 2.5, meaning the cursor positions are not updated there through the call to `whitespace()`.